src/libs/PersonalDetailsUtils.ts

```diff
@@ -0,0 +1,211 @@
+import Onyx, {OnyxEntry} from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import * as OnyxTypes from '@src/types/onyx';
+import {PersonalDetails, PersonalDetailsList} from '@src/types/onyx';
+import * as LocalePhoneNumber from './LocalePhoneNumber';
+import * as Localize from './Localize';
+import * as UserUtils from './UserUtils';
+
+let personalDetails: Array<OnyxTypes.PersonalDetails | null> = [];
+let allPersonalDetails: OnyxEntry<PersonalDetailsList> = {};
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    callback: (val) => {
+        personalDetails = Object.values(val ?? {});
+        allPersonalDetails = val;
+    },
+});
+
+/**
+ * @param [defaultValue] optional default display name value
+ */
+function getDisplayNameOrDefault(displayName?: string, defaultValue = ''): string {
+    return displayName ?? defaultValue ?? Localize.translateLocal('common.hidden');
```

Comment:
Did you test this function to ever actually fall back to `Localize.translateLocal('common.hidden')`?

`??` only checks for `undefined` and `null`, so `'' ??` will always be true.
Therefore, since you pass `defaultValue = ''`, the `defaultValue ??` check will always be `true`, and we'll return the `defaultValue` even if it's empty, won't we?